### PR TITLE
WT-12766 Dont save keys for autocommit transactions as they will never prepare

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/include/txn_inline.h [hunk 1]
```diff
@@ -115,7 +115,7 @@ __wt_txn_op_set_key(WT_SESSION_IMPL *session, const WT_ITEM *key)
     op = txn->mod + txn->mod_count - 1;
 
     if (WT_SESSION_IS_CHECKPOINT(session) || WT_IS_HS(op->btree->dhandle) ||
```
Comment (ajmorton) [inline src/include/txn_inline.h:118]: Can we add some text either here or in the function comment that this behaviour is only required when the transaction might be prepared?

**Comment by luke-pearson** [comment]
@ajmorton see c4d49b7ca.

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

File: src/include/txn_inline.h [hunk 2]
```diff
@@ -100,7 +100,8 @@ __wt_txn_op_set_recno(WT_SESSION_IMPL *session, uint64_t recno)
 
 /*
  * __wt_txn_op_set_key --
- *     Set the latest transaction operation with the given key.
+ *     Copy the given key onto the most recent transaction operation. This function early exits if
+ *     the transaction cannot prepare.
```
Comment (etienneptl) [inline src/include/txn_inline.h:104]: Why are we not mentioning the other conditions of the early exit?

**Pull request merged**